tools/generate_static_provisioning_rest_yaml.sh

```diff
@@ -0,0 +1,394 @@
+#!/bin/bash
+#
+# Copyright 2021 IBM Corp.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+usage() {
+  echo "Usage: $0
+                -f|--filesystem <Name of Volume's Source Filesystem>
+                -l|--linkpath <full Path of Volume in Primary Filesystem>
+                -F|--fileset <name of source fileset>
+                -s|--size <size in GB>
+                -u|--username <Username of spectrum scale GUI user account.>
+                -t|--password <Password of spectrum scale GUI user account.>
+                -r|--guihost <Route host name used to route traffic to the spectrum scale GUI service.>
+                [-p|--pvname <name for pv>]
+                [-c|--storageclass <StorageClass for pv>]
+                [-a|--accessmode <AccessMode for pv>]
+                [-h|--help] " 1>&2
+  exit 1
+}
+
+fullUsage() {
+  echo "Usage: $0
+                -f|--filesystem <Name of Volume's Source Filesystem>
+                -l|--linkpath <full Path of Volume in Primary Filesystem>
+                -F|--fileset <name of source fileset>
+                -s|--size <size in GB>
+                -u|--username <Username of spectrum scale GUI user account.>
+                -t|--password <Password of spectrum scale GUI user account.>
+                -r|--guihost <Route host name used to route traffic to the spectrum scale GUI service.>
+                [-p|--pvname <name for pv>]
+                [-c|--storageclass <StorageClass for pv>]
+                [-a|--accessmode <AccessMode for pv>]
+                [-h|--help]
+
+
+Example 1: Single Fileystem
```

Comment:
Lets combine the Example 1 and 2. We don't want customer to link for static volume. Use the exact path instead

Reply:
```  Warning  FailedAttachVolume  3s (x6 over 19s)  attachdetach-controller  AttachVolume.Attach failed for volume "mystaticpv2" : rpc error: code = Internal desc = ControllerPublishVolume : Volume path /mnt/fs1 is not part of the filesystem fs0
```
@deeghuge Can you specify the exact expected changes for this?

Reply:
I was thinking of something like this
```
Example 1: Directory based static volume 
This example shows how to create a volume from a directory '/ibm/fs1/staticpv' within the filesyetem 'fs1'.
	$0 --filesystem fs1 --path /ibm/fs1/staticpv --size 10 --pvname mystaticpv

Example 2: Fileset based volume
	This example shows how to create a volume from a fileset 'fileset1' within the filesyetem 'fs2'.
	$0 --filesystem fs2 --fileset fileset1 --size 10 --pvname mystaticpv
```

Also since we want to avoid softlink for static volumes, can we change option `linkpath` to `path`

Reply:
Where are you getting this error ? 
`Warning  FailedAttachVolume  3s (x6 over 19s)  attachdetach-controller  AttachVolume.Attach failed for volume "mystaticpv2" : rpc error: code = Internal desc = ControllerPublishVolume : Volume path /mnt/fs1 is not part of the filesystem fs0
`

Reply:
Got this error under events, when trying to deploy a test pod with volume mount using the generated PVC/PV. The error occurred when filesystem is `fs0` and path is `/mnt/fs1`.

Reply:
This is done, @deeghuge Kindly re-review the examples.

Reply:
>Got this error under events, when trying to deploy a test pod with volume mount using the generated PVC/PV. The error occurred when filesystem is fs0 and path is /mnt/fs1.

This is correct error as your filesystem is fs0 but path /mnt/fs1 is for different filesystem

Reply:
The current script checks that the passed path is a gpfs path or not, so in case if filesystem is `fs0` with mountpath `/mnt/fs0` however if path `/mnt/fs1` is passed along with flags, which is a gpfs path but not under fs0 mount path, the script accepts it. This was implemented considering previous example 2 using softlinks. Can you @deeghuge  help understand the requirement here, if existing implementation is fine or requires changes?

Reply:
Should the script restrict this? i.e Show error when a path outside the mount path of the filesystem is given?

Reply:
yes, if that possible